plugin/fubitive.vim

```diff
@@ -75,4 +76,8 @@ if !exists('g:fugitive_browse_handlers')
   let g:fugitive_browse_handlers = []
 endif
 
+if exists('g:fubitive_domain_context_path')
```

Comment:
I'd rather replace the `let root` construction instead of having this `if`.

Something like:
```
let root = is_cloud
    \ ? protocol . substitute(repo, ':', '/', '')
    \ : join( filter( [ protocol, domain, bitbucket_context_path, 'projects', project, 'repos', repo ], '!empty(v:val)' ), '/' )

```

Reply:
I'm in favour of eliminating this `if` as well, but I feel this suggestion makes it a bit harder to understand overall. (I think it also introduces a bug, as joining `protocol` and `domain` would result in something like `https:///bitbucket.org` - though it would be easy to fix.)

What if we made a function to return the `/`-prefixed context path or empty string?

```viml
function! s:domain_context_path()
  return exists('g:fubitive_domain_context_path')
    \ ? '/' . g:fubitive_domain_context_path
    \ : ''
endfunction
```

Reply:
> I think it also introduces a bug

Good catch!

As for simplifying instead of complicating, we could use a little bit of both approaches?
```vim
  let root = is_cloud
        \ ? substitute(repo, ':', '/', '')
        \ : join([domain . s:domain_context_path(), 'projects', project, 'repos', repo], '/')
  let root = protocol . root
```

I do still think that prepending the `protocol` on a separate line is nicer than the current approach, even if we toss out the `join()`.

Reply:
Thanks for the feedback, I've integrated your comments

Reply:
I've noticed that this still has the bug @tommcdo noticed the first time (but I'm just debugging by reading), it will join the `domain` and `/<domain_context_path>` with `/` resulting in `//`.

Perhaps  the simplest thing _would_ in fact be this:
```
        \ : domain . s:domain_context_path() . '/projects/' . project . '/repos/' . repo
```

Reply:
Alright, so you confirm it works with your company's setup of `bitbucket`?

If so, I think this is fit for merging.

Reply:
yep, it works with my company's setup :)

Reply:
@mark2185 any news here? :D

Reply:
Yup, the news is thanks for the contribution! :)